pennylane_lightning/core/src/simulators/lightning_gpu/gates/tests/Test_StateVectorCudaManaged_Param.cpp

```diff
@@ -1578,4 +1579,52 @@ TEMPLATE_TEST_CASE("LightningGPU::applyOperation multiple wires",
         CHECK(sv.getDataVector() ==
               Pennylane::Util::approx(sv_expected.getDataVector()));
     }
+}
+
+TEMPLATE_TEST_CASE("StateVectorCudaManaged::applyGlobalPhase",
+                   "[StateVectorCudaManaged_Param]", double) {
+    using ComplexT = StateVectorCudaManaged<TestType>::ComplexT;
+    std::mt19937_64 re{1337};
+    const size_t num_qubits = 3;
+    const bool inverse = GENERATE(false, true);
+    const size_t index = GENERATE(0, 1, 2);
+    const TestType param = 0.234;
+    const ComplexT phase = std::exp(ComplexT{0, (inverse) ? param : -param});
+
+    auto sv_data = createRandomStateVectorData<TestType>(re, num_qubits);
+    StateVectorCudaManaged<TestType> sv(
+        reinterpret_cast<ComplexT *>(sv_data.data()), sv_data.size());
+    sv.applyOperation("GlobalPhase", {index}, inverse, {param});
+    auto result_sv = sv.getDataVector();
+    for (size_t j = 0; j < exp2(num_qubits); j++) {
+        ComplexT tmp = phase * ComplexT(sv_data[j]);
+        CHECK((real(result_sv[j])) == Approx(real(tmp)));
+        CHECK((imag(result_sv[j])) == Approx(imag(tmp)));
+    }
+}
+
+TEMPLATE_TEST_CASE("StateVectorCudaManaged::applyControlledGlobalPhase",
+                   "[StateVectorCudaManaged_Param]", double) {
+    using ComplexT = StateVectorCudaManaged<TestType>::ComplexT;
+    std::mt19937_64 re{1337};
+    const size_t num_qubits = 3;
+    const bool inverse = GENERATE(false, true);
+    const size_t index = GENERATE(0, 1, 2);
+    // global_phase_diagonal(-np.pi/2, wires=[0, 1, 2], controls=[0, 1],
+    // control_values=[0, 1])
```

Comment:
```suggestion
```

Reply:
This is a comment saying what the `phase` diagonal is and how is it generate. I would like to keep it in case we want to change it in the future.